lib/core/finalization.toit

```diff
@@ -47,20 +47,27 @@ monitor FinalizationStack_:
     lambdas_.add lambda
     ensure_finalization_task_
 
-  wait_for_next -> bool:
+  wait_for_next -> Lambda?:
     deadline := Time.monotonic_us + IDLE_TIME_MS * 1_000
-    return try_await --deadline=deadline: not lambdas_.is_empty
+    try_await --deadline=deadline: not lambdas_.is_empty
+    // If we got a lambda, we must return it and deal with even if we timed out.
+    return lambdas_.is_empty ? null : lambdas_.remove_last
 
   ensure_finalization_task_ -> none:
```

Comment:
Name feels a bit off. Sounds like it's a field/getter for a task. However, it is running/creating something instead.

Reply:
It's a valid point.